src/main/kotlin/no/nav/familie/ef/søknad/api/dto/Fødselsnummer.kt

```diff
@@ -0,0 +1,59 @@
+package no.nav.familie.ef.søknad.api.dto
+
+import java.time.LocalDate
+
+class Fødselsnummer(val verdi: String) {
+
+    val erDNummer = verdi.substring(0, 1).toInt() > 3
+    val fødselsdato: LocalDate
+
+    init {
+        check(gyldig())
+        fødselsdato = beregnFødselsdato()
+    }
+
+    private fun beregnFødselsdato(): LocalDate {
+        val dag = verdi.substring(0, 2).toInt() - (if (erDNummer) 30 else 0)
+        val måned = verdi.substring(2, 4).toInt()
+        val år = verdi.substring(4, 6).toInt()
+        val datoUtenÅrhundre = LocalDate.of(år, måned, dag)
+        val individnummer = verdi.substring(6, 8).toInt()
+        when {
+            individnummer in 0..499 -> return datoUtenÅrhundre.plusYears(1900)
+            individnummer in 500..749 && år >= 54 && år <= 99 -> return datoUtenÅrhundre.plusYears(1800)
+            individnummer in 900..999 && år >= 40 && år <= 99 -> return datoUtenÅrhundre.plusYears(1900)
+            individnummer in 500..999 && år >= 0 && år <= 39 -> return datoUtenÅrhundre.plusYears(2000)
+        }
+        throw IllegalArgumentException()
+    }
+
+    private fun gyldig(): Boolean {
+        if (verdi.length != 11 || verdi.toLongOrNull() == null) {
+            return false
+        }
+
+        val siffer = verdi.chunked(1).map { it.toInt() }
```

Comment:
Bra jobba, Bent!

Et kotlin-spm bare: hvorfor kan vi ikke gjøre `toInt` direkte på `verdi` her?

Reply:
Chunked returnerer en liste av strings med størrelse gitt av parameter til chunked. I dette tilfellet 1. Hvert enkelt siffer gjøres deretter om til int for utregningens skyld, slik at jeg ender opp med en liste med elleve inter.  Hvis jeg hadde brukt toInt direkte ville jeg hatt et tall på 10 eller 11 siffer som hadde vært kronglete å plukke ett og ett siffer fra.